rpc.go

```diff
@@ -316,7 +292,7 @@ func (l *Libvirt) deregister(id uint32) {
 // request performs a libvirt RPC request.
 // The returned channel is used by the caller to receive the asynchronous
 // call response. The channel is closed once a response has been sent.
-func (l *Libvirt) request(proc uint32, program uint32, payload *bytes.Buffer) (<-chan response, error) {
+func (l *Libvirt) request(proc uint32, program uint32, payload *bytes.Buffer) (response, error) {
```

Comment:
This should probably return (*response, error) - the response can be quite large and we don't want to copy it.

Reply:
The response struct is just a couple of words big: it contains only uint32 and slice (ptr to data + 2 ints). I figured there's no real benefit in storing this in heap. Besides we were copying this data to and from channel before anyway.